content/en/hosting/cht/migration/preparing-for-5.md

```diff
@@ -0,0 +1,161 @@
+
+Additionally, the web developer console will show:
+
+```shell
+Error selecting contact Error: Rules Engine: Rules are not declarative. Updates are required.
+ERROR Error: Uncaught (in promise): TypeError: Cannot read properties of null (reading 'doc')
+TypeError: Cannot read properties of null (reading 'doc')
+```
+
+The fix is to:
+1. Update [CHT Conf](/community/contributing/code/cht-conf/) to be current
+2. Recompile the app settings and upload them to the CHT. Be sure to replace the value of `--url` to the current password, URL and port: `cht --url=https://medic:password@example.com:12345 compile-app-settings upload-app-settings`
+
+
+Background information:
+* [Make declarative config mandatory](https://github.com/medic/cht-core/issues/5906)
+
+### Increase ecmaVersion ES linting to version 6
+
+{{< callout type="info" >}} Applies to: CHT versions between 4.0 and 4.4  {{< /callout >}}
+
+Deployments running CHT Version 4.0 and 4.4 must upgrade to version 4.5 or later before upgrading to version 5.x.
+
+Background information:
+* [Increase ecmaversion linting for ddocs](https://github.com/medic/cht-core/issues/9202)
+
+### Ensure booleans are used with signoff fields
+
+{{< callout type="info" >}} Applies to: Instances where `needs_signoff` triggers replication of reports  {{< /callout >}}
+
+In some scenarios where `needs_signoff` is set to string `"false"` instead of boolean `false`, replication will happen when it shouldn't.  This is because the string of `"false"` is a truthy value.  
+
+CHT 5.0 now correctly enforces this to be a boolean.  Deployments to make sure that workflows that use `needs_signoff` field have a boolean instead of a string.
+
+Background information:
+* [Replication fails to filter out reports with `needs_signoff` set to false](https://github.com/medic/cht-core/issues/10183)
+
+### Angular 20 requires Chrome 107 or later
+
+{{< callout type="info" >}} Applies to: all instances  {{< /callout >}}
+
+CHT 5.0 is preparing fo an upgraded version of [Angular](https://blog.angular.dev/announcing-angular-v20-b5c9c06cf301). As this version of Angular will require Chrome 107 or later, we're making this a requirement in 5.0 as a breaking change to pave the path for the upgrade later. When [researching the impact of this upgrade](https://github.com/medic/cht-core/issues/10029#issuecomment-3358338361) we surveyed over 100,000 end users devices and found that less than 0.5% of users are affected.  These users are running Chrome 106 released 3 years ago at this writing. A best practice is to have users always run the latest version if possible.
+
+To check if any of your users are impacted, check the [user-devices API](/building/reference/api/#get-apiv2exportuser-devices). This is an authenticated API endpoint and will return JSON for all users for all time.  As this may include multiple entries per user, it's important to filter out duplicate users and of course filter out users who are running Chrome 107 or later. 
```

Comment:
```suggestion
To check if any of your users are impacted, check the [user-devices API](/building/reference/api/#get-apiv2exportuser-devices). This is an authenticated API endpoint and will return JSON for all users for all time.  As this may include multiple entries per user, it's important to filter out duplicate users and those running Chrome 107 or later. 
```